roles/capi_cluster/defaults/main.yml

```diff
@@ -466,6 +476,28 @@ capi_cluster_release_defaults:
                 else {}
               )
           }}
+  etcd: >-
+    {{-
+      { "blockDevice": { "size": capi_cluster_etcd_blockdevice_size } } |
+        combine(
+          { "blockDevice": { "type": "Local" } }
+          if capi_cluster_etcd_blockdevice_type | capitalize  == "Local"
+          else 
+          { "blockDevice": { "type": "Volume" } }
+        ) |
+        combine(
+          { "blockDevice": { "volumeType": capi_cluster_etcd_blockdevice_volume_type } }
+          if capi_cluster_etcd_blockdevice_volume_type | len > 0
+          else {}
+        ) |
+        combine(
+          { "blockDevice": { "availabilityZone": capi_cluster_etcd_blockdevice_volume_az } }
+          if capi_cluster_etcd_blockdevice_volume_az | len > 0
+          else {}
+        )
+      if capi_cluster_etcd_blockdevice_size > 0
+      else {}
+    }}
```

Comment:
If we set the defaults above, this can become

```
etcd: >-
  {{-
    {
      "blockDevice": (
        {
          "size": capi_cluster_etcd_blockdevice_size,
          "type": capi_cluster_etcd_blockdevice_type
        } |
          combine(
            { "volumeType": capi_cluster_etcd_blockdevice_volume_type }
            if capi_cluster_etcd_blockdevice_volume_type
            else {}
          ) |
          combine(
            { "availabilityZone": capi_cluster_etcd_blockdevice_volume_az }
            if capi_cluster_etcd_blockdevice_volume_az
            else {}
          )
      )
    }
    if capi_cluster_etcd_blockdevice_size > 0
    else {}
  }}
```